Instructions for sideloading on Firefox for Android

rebloor opened the pull request:
- Removes the term sideloading (and replaces with "manual installation")
- Adds instructions to the (now) installing self-distributed add-ons page on how to activate and use **Install Extension from File** on Firefox for Android. 
- Completed various minor edits to the affected pages.

Fixes #2028

dotproto commented on src/content/documentation/publish/distribute-sideloading.md:
The naming of this article is unfortunate as this term has a specific meaning. As described in [Support for extension sideloading has ended](https://blog.mozilla.org/addons/2020/03/10/support-for-extension-sideloading-has-ended/), "sideloading" generally refers to an installation method where installing an application on your system would also result in adding and extension to Firefox. 

The "Installation using the standard extension folders" section does call out that this feature is no longer supported, but the intro still seems to suggest it is supported.

rebloor replied:
@dotproto Isn't sideloading a general term meaning  "installing applications or software on a device from sources other than the official app store or distribution channels, bypassing the usual vetting process". The feature removed was the ability to sideload "without the user taking an action"?

rebloor replied:
@dotproto I have revisited and removed the information about installing with an app. Given this was removed five years ago, there seems a little point in including the documentation.

dotproto commented on src/content/documentation/publish/distribute-sideloading.md:
```diff
@@ -15,13 +15,13 @@ date: 2019-11-06 04:05:42
 
 ::: note
-No automatic updates will be performed for add-ons installed using these methods. You will have to deliver a new XPI file to your user for each update. However, automatic compatibility checks are still performed.
+No automatic updates are performed for add-ons installed using these methods. You have to deliver a new XPI file to your user for each update. However, automatic compatibility checks are performed.
```
I don't think this is accurate. I tried walking through extensions source files on searchfox but I didn't see anything that suggested updates are handled different based on installation method.

@willdurand, can you confirm?

rebloor replied:
@dotproto after reviewing the documentation, I believe this statement was actually intended to refer to extensions offered as a beta update release to a listed extension. I've updated the note.

dotproto commented on src/content/documentation/publish/distribute-sideloading.md:
```diff
@@ -67,25 +67,25 @@ An email address style ID is recommended.
+## Install add-on from file on a computer
 
-To use Install Add-on From File in Add-on Manager, send the user the signed add-on with the following instructions:
+To sideload an add-on in desktop Firefox, send the user the signed add-on with these instructions:
```
We should avoid the term "sideload" here as that has a more specific meaning.

```suggestion
To install an add-on from a file in desktop Firefox, send the user the signed add-on with these instructions:
```

rebloor commented:
@dotproto @willdurand I have now adjusted the content to remove the term sideloading. I've also renamed the page's md file and created https://github.com/mozilla/extension-workshop/issues/2045 for the introduction of a redirect.

rebloor commented on src/content/documentation/publish/self-distribution.md:
```diff
@@ -89,15 +98,15 @@ When distributing an add–on from your website, you have two options for making
 
 ::: note
-You may find references to using the `InstallTrigger` API to trigger a download from JavaScript. This API was never officially supported and will be removed from Firefox 102. Use is therefore not recommended.
+You may find references to using the `InstallTrigger` API to trigger a download from JavaScript. This API was never officially supported and will be removed from Firefox 102. Use is, therefore, not recommended.
```
@dotproto it's not been removed, and the relevant bug is still open.
```suggestion
You may find references to using the `InstallTrigger` API to trigger a download from JavaScript. This API is not supported and it will be removed (see [Bug 1776426](https://bugzilla.mozilla.org/show_bug.cgi?id=1776426)).  Use is, therefore, not recommended.
```
